### PR TITLE
feat(board): add puchi_ble_v1 to boards

### PR DESCRIPTION
**Pull request by BenRoe**
## Board/Shield Check-list
 - [x] This board/shield is tested working on real hardware
 - [x] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [x] `.zmk.yml` metadata file added
 - [ ] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [ ] General consistent formatting of DeviceTree files
 - [ ] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [ ] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [ ] If split, no name added for the right/peripheral half
 - [ ] Kconfig.defconfig file correctly wraps *all* configuration in conditional on the shield symbol
 - [ ] `.conf` file has optional extra features commented out
 - [ ] Keyboard/PCB is part of a shipped group buy or is generally available in stock to purchase (OSH/personal projects without general availability should create a zmk-config repo instead)

Hi,
my name is Ben from Keycapsss.com. The Puchi-BLE is a nrfMicro 1.3 fork.
I am in contact with @joric (creator of the nrfMicro). He is fine with the fork.
The board is tested and works.

I have encapsulate board (nrfmicro_13 -> puchi_ble_v1) for future changes.
I am not sure if all files have proper Copyright + License headers. If something is missing, let me know and i will change it.

### CODE REVIEW

File: app/boards/arm/puchi_ble/Kconfig [hunk 1]
```diff
@@ -0,0 +1,10 @@
+config BOARD_ENABLE_DCDC
+	bool "Enable DCDC mode"
+	select SOC_DCDC_NRF52X
+	default y
```
Comment (petejohanson) [inline app/boards/arm/puchi_ble/Kconfig:4]: This is already the default, this line can be omitted.

File: app/boards/arm/puchi_ble/Kconfig [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+	depends on (BOARD_PUCHI_BLE_v1)
+
+config BOARD_PUCHI_BLE_CHARGER
+	bool "Enable battery charger"
+	default y
```
Comment (petejohanson) [inline app/boards/arm/puchi_ble/Kconfig:9]: Ditto.

File: app/boards/arm/puchi_ble/Kconfig.board [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Puchi-BLE board configuration
+
+# Copyright (c) 2020 The ZMK Contributors
```
Comment (petejohanson) [inline app/boards/arm/puchi_ble/Kconfig.board:3]: Update the date please.

File: app/boards/arm/puchi_ble/Kconfig.defconfig [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Electronut Labs Papyr board configuration
```
Comment (petejohanson) [inline app/boards/arm/puchi_ble/Kconfig.defconfig:1]: Update to match?

File: app/boards/arm/puchi_ble/Kconfig.defconfig [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+config ZMK_USB
+	default y
+
+config PINMUX
+	default y
+
+if BOARD_PUCHI_BLE_v1
+
+config BOARD_PUCHI_BLE_CHARGER
```
Comment (petejohanson) [inline app/boards/arm/puchi_ble/Kconfig.defconfig:32]: I don't believe this section is necessary, board specific Kconfig will automatically get included.

**Review by petejohanson — CHANGES_REQUESTED** [review]
Please be sure to check all the copyright and license headers on every file.

Thanks!

**Comment by BenRoe** [comment]
@petejohanson all requested code changes should be fixed now.

File: app/boards/arm/puchi_ble/arduino_pro_micro_pins_flipped.dtsi [hunk 6]
```diff
@@ -0,0 +1,57 @@
+/*
```
Comment (petejohanson) [inline app/boards/arm/puchi_ble/arduino_pro_micro_pins_flipped.dtsi:1]: Missed this the first time... Are you actually using this flipped file at all? Seems unnecessary.

**Review by petejohanson — COMMENTED** [review]
Just one outstanding question.

File: app/boards/arm/puchi_ble/arduino_pro_micro_pins_flipped.dtsi (hunk 6 above)
Comment (BenRoe) [reply app/boards/arm/puchi_ble/arduino_pro_micro_pins_flipped.dtsi]: You are right, i deleted it.

**Pull request merged**